cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -65,19 +66,23 @@ public class KafkaBrokerConfigurationDiff extends AbstractJsonDiff {
             + "|zookeeper\\.clientCnxnSocket"
             + "|broker\\.rack)$");
 
+    /**
+     * KRaft controller configuration options are skipped if it is not combined node
+     */
+    private static final Pattern IGNORABLE_CONTROLLER_PROPERTIES = Pattern.compile("controller\\.quorum\\..*");
```

Comment:
This is so that the brokers don't roll for configuration changes that are actually only related to the controllers. Since it doesn't make sense to include them when calculating the diff. I think this should protect us in future if any of the configurations are marked as dynamically configurable.

Reply:
The `controller.listener.names` config applies to both brokers and controllers. So we shouldn't ignore it here because we want the brokers to notice that this value has changed

Reply:
You are absolutely right!